### ISSUE TITLE
Create-FhirServerSamplesEnvironment.ps1 fails when using 'fhirSweden' as environment name

### ISSUE THREAD

**Issue by jwalter** [microsoft/fhir-server-samples#48]
I could not get the script to complete when using an environment name that had 11 chars but changing it to a name with 7 chars worked.

```
New-AzureRmResourceGroupDeployment : 8:58:01 AM - Resource Microsoft.HealthcareApis/services 'fhirSandbox' failed with message '{
  "error": {
    "code": "BadRequest",
    "message": "Name can contain only upper and lowercase letters, numbers and the '-' character."
  }
}'
At /home/joakim/fhir-server-samples/deploy/scripts/Create-FhirServerSamplesEnvironment.ps1:128 char:1
+ New-AzureRmResourceGroupDeployment -TemplateUri $sandboxTemplate -env ...
```

**Comment by hansenms** [comment]
@jwalter : Thanks for the feedback. However, I am unable to reproduce, I just created one with an 11 character name. Do you have more details on which region you were provisioning in, etc. There is some chance that this was a transient problem with the name checking or something.

**Comment by jwalter** [comment]
@hansenms : I am using the 'ukwest' region for this. The name I am trying to use is 'fhirSweden' which is actually 10 characters. I've retried it this morning with the same result; 'fhirSweden' does not work but 'fhirswe' does.

**Comment by hansenms** [comment]
How about `fhirsweden` (lower case). We don't allow upper case letters in the service name. 


![image](https://user-images.githubusercontent.com/1149740/58034702-0b26d980-7adc-11e9-987a-c7ef96cb010c.png)

**Comment by jwalter** [comment]
Ok, then it's just the error message you get when using the ps1 script that is wrong:
`Name can contain only upper and lowercase letters`
and I suppose that message is created by some component that is not part of this repository?

**Comment by JimFett** [comment]
@mihansen, this is similar to the message I was getting a few weeks ago.  @jawalte how long have you had your subscription?  I had similar issues but had to explicitly add myself as a role to the sub/RG that had enough permissions to perform CRUD operations.  My sub had been around so long that I was a Service Administrator under the IAM "Classic Administrators" but I had no explicit rights at the subscription level.  I added myself with an appropriate role and the deployment worked.  Don't know if this is the case with you, James, but its worth checking out.

**Comment by hansenms** [comment]
I can add a check in the create script for the repo. I will put it on the todo list. We should have a better check before we start the template deployment.
